### DISCUSSION TITLE
Understanding mqtt topics (with regards to upgrading from zwave2mqtt)

### DISCUSSION THREAD

**Discussion by mrherman** [zwave-js/zwave-js-ui#1282] [Q&A]
I am upgrading from zwave2mqtt to zwavejs2mqtt.  I've noticed that the way zwavejs2mqtt handles mqtt topics is different.  For example for my Zen15 device I used to have /node/50/0/1 which is now changed to /node/50/0/value/66561

I would like to understand the new topics better (both the last number and why sometimes it is 'value' and sometimes 'currentValue' with no number after it). Is it predictable (between different devices/manufactures) or mostly specific to each device?  I need to go through my previous code (made to work with zwave2mqtt) and update the mqtt topics from the old values to the new values, I am hoping if I understand them I can find a way to optimize updating them.

**Comment by robertsLando** [comment]
@mrherman topics are differents essentially because zwave-js uses a different way to unique identify a valueid.

In OZW we had nodeId-instance-CC-index now we have nodeId-endpoint(like the old instance)-CC-property-propertyKey. `property` and `propertyKey` can both be string or a number, it depends on the CC/Value, also propertyKey can be undefined. There have been many discussion about this but I find it much better then OZW method for many reasons. Essentially all CC will have the same `properties` around all the nodes so it doesn't depend on the device
